### PR TITLE
[AzureMonitorDistro] add tests for httpclient instrumentation

### PR DESCRIPTION
**Pull request by TimothyMothra**
## Changes
- add tests in new class `HttpClientInstrumentationTests`
  - using `[Collection("InstrumentationLibraries")]` to disable parallelization.
  - update `SqlClientTests` to use same collection name.
- added extension method in class `ServiceCollectionExtensions` to simplify configuration

### CODE REVIEW

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/tests/Azure.Monitor.OpenTelemetry.AspNetCore.Tests/E2ETests/HttpClientInstrumentationTests.cs [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+            // The TransmitterFactory is invoked by the Exporter during initialization to ensure that there's only one instance of a transmitter/connectionString shared by all Exporters.
+            // Here we're setting that instance to use the MockTransmitter so this test can capture telemetry before it's sent to Azure Monitor.
+            Exporter.Internals.TransmitterFactory.Instance.Set(connectionString: testConnectionString, transmitter: mockTransmitter);
+
+            // SETUP OPENTELEMETRY WITH AZURE MONITOR DISTRO
+            var activities = new List<Activity>();
+            var serviceCollection = new ServiceCollection();
+
+            // This shouldn't be needed but Http Instrumentation library was performing redaction without it.
+            serviceCollection.AddEnvironmentVariables(new Dictionary<string, string?> { { "OTEL_DOTNET_EXPERIMENTAL_HTTPCLIENT_DISABLE_URL_QUERY_REDACTION", "true" } });
```
Comment (TimothyMothra) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/tests/Azure.Monitor.OpenTelemetry.AspNetCore.Tests/E2ETests/HttpClientInstrumentationTests.cs:60]: I don't know why this is necessary.
When I debugged the Distro, this value was always set correctly.
But explicitly setting this, the Instrumentation library was redacting the query string

**Comment by TimothyMothra** [comment]
@rajkumar-rangaraj this is ready for review

**Review by rajkumar-rangaraj — APPROVED** [review]

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/tests/Azure.Monitor.OpenTelemetry.AspNetCore.Tests/E2ETests/HttpClientInstrumentationTests.cs (hunk 1 above)
Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/tests/Azure.Monitor.OpenTelemetry.AspNetCore.Tests/E2ETests/HttpClientInstrumentationTests.cs:60]: Now that we've solved the other concurrency issues, need to revisit this. May have been a side effect of the other tests.

Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/tests/Azure.Monitor.OpenTelemetry.AspNetCore.Tests/E2ETests/HttpClientInstrumentationTests.cs:60]: Still broken. will investigate this separately

**Pull request merged**